server/src/main/java/org/elasticsearch/transport/RemoteClusterPortSettings.java

```diff
@@ -0,0 +1,175 @@
+        REMOTE_CLUSTER_PREFIX + "publish_port",
+        -1,
+        -1,
+        Setting.Property.NodeScope
+    );
+
+    public static final Setting<Boolean> TCP_KEEP_ALIVE = boolSetting(
+        REMOTE_CLUSTER_PREFIX + "tcp.keep_alive",
+        TransportSettings.TCP_KEEP_ALIVE,
+        Setting.Property.NodeScope
+    );
+
+    public static final Setting<Integer> TCP_KEEP_IDLE = intSetting(
+        REMOTE_CLUSTER_PREFIX + "tcp.keep_idle",
+        TransportSettings.TCP_KEEP_IDLE,
+        -1,
+        300,
+        Setting.Property.NodeScope
+    );
+
+    public static final Setting<Integer> TCP_KEEP_INTERVAL = intSetting(
+        REMOTE_CLUSTER_PREFIX + "tcp.keep_interval",
+        TransportSettings.TCP_KEEP_INTERVAL,
+        -1,
+        300,
+        Setting.Property.NodeScope
+    );
+
+    public static final Setting<Integer> TCP_KEEP_COUNT = intSetting(
+        REMOTE_CLUSTER_PREFIX + "tcp.keep_count",
+        TransportSettings.TCP_KEEP_COUNT,
+        -1,
+        Setting.Property.NodeScope
+    );
+
+    public static final Setting<Boolean> TCP_NO_DELAY = boolSetting(
+        REMOTE_CLUSTER_PREFIX + "tcp.no_delay",
+        TransportSettings.TCP_NO_DELAY,
+        Setting.Property.NodeScope
+    );
+
+    public static final Setting<Boolean> TCP_REUSE_ADDRESS = boolSetting(
+        REMOTE_CLUSTER_PREFIX + "tcp.reuse_address",
+        TransportSettings.TCP_REUSE_ADDRESS,
+        Setting.Property.NodeScope
+    );
+
+    public static final Setting<ByteSizeValue> TCP_SEND_BUFFER_SIZE = Setting.byteSizeSetting(
+        REMOTE_CLUSTER_PREFIX + "tcp.send_buffer_size",
+        TransportSettings.TCP_SEND_BUFFER_SIZE,
+        Setting.Property.NodeScope
+    );
+
+    public static final Setting<ByteSizeValue> TCP_RECEIVE_BUFFER_SIZE = Setting.byteSizeSetting(
+        REMOTE_CLUSTER_PREFIX + "tcp.receive_buffer_size",
+        TransportSettings.TCP_RECEIVE_BUFFER_SIZE,
+        Setting.Property.NodeScope
+    );
+
+    static void validateRemoteAccessSettings(Settings settings) {
+        if (REMOTE_CLUSTER_PORT_ENABLED.get(settings)
+            && settings.getGroups("transport.profiles.", true).keySet().contains(REMOTE_CLUSTER_PROFILE)) {
+            throw new IllegalArgumentException(
+                "Remote Access settings should not be configured using the ["
+                    + REMOTE_CLUSTER_PROFILE
+                    + "] profile. "
+                    + "Use the ["
+                    + REMOTE_CLUSTER_PREFIX
+                    + "] settings instead."
+            );
+        }
+    }
+
+    public static TcpTransport.ProfileSettings buildRemoteAccessProfileSettings(Settings settings) {
+        validateRemoteAccessSettings(settings);
+
+        // Build a synthetic settings object with the `_remote_access` profile properly configured per the friendlier settings,
+        // but inherit any `_remote_access` profile settings that don't conflict.
```

Comment:
This comment was left over from a previous version that allowed using the profile settings as long as they didn't conflict, but I decided that was 1) really ugly in code, and 2) unnecessary to support because why would you want to do that. I'll fix the content and convert this to a Javadoc comment.

Reply:
I deleted the comment since it is not what we intended to do.